### PR TITLE
feat: add bulk purge api

### PR DESCRIPTION
**Pull request by pedrokiefer**

### CODE REVIEW

File: api/cache.go [hunk 1]
```diff
@@ -36,3 +37,37 @@ func cachePurge(c echo.Context) error {
...
+		count, err := manager.PurgeCache(c.Request().Context(), name, args)
+		if err != nil {
+			r = rpaas.PurgeCacheBulkResult{Path: args.Path, Error: err.Error()}
+		} else {
+			r = rpaas.PurgeCacheBulkResult{Path: args.Path, InstancesPurged: count}
+		}
+		results = append(results, r)
+	}
+
+	return c.JSON(http.StatusOK, results)
```
Comment (wpjunior) [inline api/cache.go:72]: @pedrokiefer could you change to emit 500 when some error occur ?

**Review by wpjunior — CHANGES_REQUESTED** [review]

File: api/cache.go (hunk 1 above)
Comment (pedrokiefer) [reply api/cache.go:72]: I can, but does it really makes sense? Say I'm asking to purge 10 URLs, only one fails, should it really be 500?

Comment (wpjunior) [reply api/cache.go:72]: it is important to we detect that have some problem with that route, we could use metrics to achieve that. @pedrokiefer

File: api/cache.go [hunk 2]
```diff
@@ -36,3 +37,39 @@ func cachePurge(c echo.Context) error {
...
+	if len(name) == 0 {
+		return c.String(http.StatusBadRequest, "instance is required")
+	}
+	manager, err := getManager(c)
+	if err != nil {
+		return err
+	}
+
+	var argsList []rpaas.PurgeCacheArgs
+	if err = json.NewDecoder(c.Request().Body).Decode(&argsList); err != nil {
```
Comment (cezarsa) [inline api/cache.go:56]: Any reason on why not to use c.Bind() and let echo handle parsing based on content-type?

Comment (pedrokiefer) [reply api/cache.go:56]: This reason here: https://github.com/labstack/echo/issues/1356 :(

Comment (cezarsa) [reply api/cache.go:56]: ugh

**Review by cezarsa — APPROVED** [review]

**Review by nettoclaudio — APPROVED** [review]
lgtm

**Pull request merged**